components/merkledb/src/views/address.rs

```diff
@@ -289,17 +289,25 @@ pub struct ResolvedAddress {
     /// for different views in the same column family. In other words, key spaces for two addresses
     /// with equal `name` and `id`s `Some(x)` and `Some(y)`, `x != y`, must not intersect.
     pub id: Option<NonZeroU64>,
+
+    /// No-op field for forward compatibility.
+    non_exhaustive: (),
```

Comment:
I'd argue that breaking `ResolvedAddress` structure will lead to major breakage in MerkleDB, so this seems overly defensive. Also, the structure is only public because of its use in `Snapshot`, which - let's face it - we don't intend for third parties to implement.